Modules/Registration/Common/include/itkImageToImageMetric.hxx

```diff
@@ -1329,7 +1329,7 @@ ImageToImageMetric< TFixedImage, TMovingImage >
 ::GetValueAndDerivativeMultiThreadedPostProcessInitiate() const
 {
   m_Threader->SetSingleMethod( GetValueAndDerivativeMultiThreadedPostProcess,
-                               (void *)( &m_ThreaderParameter ) );
+                               (const void *)( &m_ThreaderParameter ) );
```

Comment:
I would suggest to avoid any old-school C-style cast. The warning can probably be fixed by `const_cast<void*>`. Of course, it is still risky to cast away the `const`.